Common/src/Pollster/AgentHandler.cs

```diff
@@ -0,0 +1,108 @@
+﻿// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2021-$CURRENT_YEAR$. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+using ArmoniK.Core.Common.gRPC.Services;
+using ArmoniK.Core.Common.Utils;
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.DependencyInjection;
+
+using Serilog;
+
+namespace ArmoniK.Core.Common.Pollster;
+
+public class AgentHandler : IAgentHandler, IAsyncDisposable
+{
+  private readonly IAgent          agent_;
+  private readonly LoggerInit      loggerInit_;
+  private          WebApplication? app_;
+
+  public AgentHandler(IAgent  agent,
+                      LoggerInit loggerInit)
+  {
+    agent_           = agent;
+    loggerInit_ = loggerInit;
+  }
+
+  public Task Start(string sessionId,
+                    string taskId,
+                    string socketPath)
+  {
+    agent_.Init(sessionId,
+                taskId);
+
+    var builder = WebApplication.CreateBuilder();
+
+    builder.Services.AddSingleton(agent_)
+           .AddLogging(loggerInit_.Configure)
+           .AddGrpc();
+
+    builder.WebHost.ConfigureKestrel(options =>
+                                     {
+                                       if (File.Exists(socketPath))
```

Comment:
This is useless and will hide issues.

Reply:
the deletion of the file ?

Reply:
Yes. On Linux, you can delete a file in use by another process, and the other process will keep the file opened nonetheless, but the file will no longer be accessible by its path.
Therefore, when a new socket will be created, a new file with a new inode will be created, and you will be able to bind to it even in the case where another process was still listening on the previous socket.
Without deleting the socket file, I expect the server to crash when it tries to listen to it while another process is still listening on it.